Add GITOID_URI_SPEC (#7)
git-bom/spec: GITOID_URI_SPEC.md

Scheme name: gitoid

Status: provisional

Applications/protocols that use this scheme name:
     See [Section 3.5](https://www.rfc-editor.org/rfc/rfc7595.html#section-3.5).
Contact:
     Registering party: Ed Warnicke <eaw@cisco.com>

Scheme Creator: GitBOM

Change controller:

>     Either the registering party or someone who is verified to represent

Reviewer: Should reword to: "Change controller: Either the registering party or someone verified to represent the scheme creator. See the previous answer."

Author: I believe I've fixed this.

    the scheme creator.  See previous answer.

References:
https://git-scm.com/book/en/v2/Git-Internals-Git-Objects

Scheme syntax: gitoid":"<git object type>":"<hash algorithm>":"<hash value>

Current valid values for <git object type> are 'blob','tree','commit', and 'tag'.
Current valid values for <hash algorithm> are 'sha1' and 'sha256'

> <hash value> should be expressed as a hexidecimal string in lower case.

Reviewer: s/hexidecimal/hexadecimal/

Author: I believe I've fixed this.

Example: gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64

> 

Reviewer 2: would it be worth noting here that, at present, our schema only recognizes using one of the four types ('blob') or that we may add additional types (e.g., 'bom')? or do you think this information should be captured elsewhere?

Author: I think we put this in as a general schema for gitoids, which we then use.  Not as a specific gitbom schema.

Scheme semantics:

> gitoid stands for Git Object ID.  Git is an object store.  It currently has four types of objects.  A git object is formed by prepending an object header to the contents of the object.  A git object header consists of:

Reviewer 2: ```suggestion
gitoid stands for Git Object ID.  Git is an object store, which currently supports four types of objects.  A git object is formed by prepending an object header to the contents of the object.  A git object header consists of:
```

Author: I believe I've fixed this.

> gitoid stands for Git Object ID.  Git is an object store.  It currently has four types of objects.  A git object is formed by prepending an object header to the contents of the object.  A git object header consists of:

Reviewer: s/contents of the object/object's contents/

Author: I believe I've fixed this.

<git object type>" "<size of contents as decimal string>"\0"

> The git object id is computed by applying a hash to the git object.  Currently the most common hash is sha1, but there is some effort to transition to sha256 over time.

Reviewer: s/Currently/Currently,/

Reviewer: Also, this line may end up changing... e.g. if we go with the unified line where we have more than one hash. Might be `sha1+sha256` or some similar derivative

Author: > s/Currently/Currently,/

I believe I've fixed this.

Author: > Also, this line may end up changing... e.g. if we go with the unified line where we have more than one hash. Might be `sha1+sha256` or some similar derivative

Let's leave this be for the moment.
